Sources/Features/Breadcrumb/BacktraceBreadcrumbsLogManager.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+@objc class BacktraceBreadcrumbsLogManager: NSObject {
+    
+    private var breadcrumbId = Date().millisecondsSince1970
+    
+    private var backtraceBreadcrumbFileHelper: BacktraceBreadcrumbFileHelper?
+
+    init(_ breadcrumbLogPath: String, maxQueueFileSizeBytes: Int) {
+        super.init()
+        self.backtraceBreadcrumbFileHelper = BacktraceBreadcrumbFileHelper(breadcrumbLogPath, maxQueueFileSizeBytes: maxQueueFileSizeBytes)
+    }
+    
+    func addBreadcrumb(_ message: String,
+                       attributes:[String:Any]? = nil,
+                       type: BacktraceBreadcrumbType,
+                       level: BacktraceBreadcrumbLevel) -> Bool {
+        let time = Date().millisecondsSince1970
+        var breadcrumb: [String: Any] = ["timestamp": time,
+                                         "id": breadcrumbId,
+                                         "level": level.info,
+                                         "type": type.info,
+                                         "message": message]
+        if let attributes = attributes, !attributes.keys.isEmpty {
```

Comment:
do we need to have this for loop if we already have attributes in the format [String:Any]? Also - shouldn't we expect attributes in format `[String: String]` ?

Reply:
done